### ISSUE TITLE
Object hierarchy

### ISSUE THREAD

**Issue by betterapp** [BlackbitDigitalCommerce/pimcore-data-director#36]
`return 'ClassName:fieldName:'.$params['value'];`

What it will return if object will be not found ?
Is it possible that it will return bool value ?

Documentation say: DD try to create that object - is that right ?

**Comment by BlackbitNeueMedien** [comment]
For all fields except `Path` it will return the object (of found) or `false` if not found.
Only for field `Path` this object will get created (because classname is known and in the resulting object this one filter field which got used will get populated, object is unpublished). This is mainly for when you have imported the data in the wrong order - when we would not create those dummy parent elements, the import would fail. The Data Director usually tries to be as tolerant concerning human mistakes as possible.

You can see this behaviour when putting 
`return 'Product:key:non-existing-key:json_encode';
somewhere.

`But what if it can not create it ?`
Then there is a warning log and the raw data item gets skipped.

**Comment by betterapp** [comment]
I think if it will not find it should return NULL not false;

**Comment by BlackbitNeueMedien** [comment]
Hehe, yes, also wondered about this. Perhaps will change this, but didn't you ask `Is it possible that it will return bool value ?`? ;-)
Will check where this boolean comes from...

**Comment by betterapp** [comment]
;)

**Comment by betterapp** [comment]
Is it possible that return false is only on DD 3.X ?
We did not see this problem on 2.8

**Comment by BlackbitNeueMedien** [comment]
Might be, as far as I know formerly it returned the string "ClassName:fieldName:abc" (or whatever value `$params['value']` has). Or what have you got in 2.8?

**Comment by betterapp** [comment]
I do not know because we updated to 3.X ;)

**Comment by BlackbitNeueMedien** [comment]
Fixed it in 3.1. For `Product:key:non-existing:gettype` you now get `NULL`.

**Comment by betterapp** [comment]
but there is no tag for 3.1 yet

**Comment by BlackbitNeueMedien** [comment]
Yes, because it is not released yet. You can use 3.1.x-dev but try on a test instance first.
